chore: Remove all deprecated using directive styles

tgodzik opened the pull request:
It seems that we still support multiple ways of declaring using directives, which should be removed before 1.0.x.

It's also needed for SIP so that it's clear what can be accepted by the SIP commitee.

tgodzik committed:
chore: Update using directives to a version that doesn't support multiline

lwronski reviewed (commented):
Can you also remove invalid example from [documentation](https://scala-cli.virtuslab.org/docs/guides/using-directives#semantics)?

tgodzik committed:
improvement: Support only one type of using directives

It seems that we still support multiple ways of declaring using directives, which should be removed before 1.0.x.

It's also needed for SIP so that it's clear what can be accepted by the SIP commitee.

tgodzik commented:
> Can you also remove invalid example from [documentation](https://scala-cli.virtuslab.org/docs/guides/using-directives#semantics)?

I realized we have some more invalid syntax, so trying to update it altogether. Not sure which tests will fail, so will need to check on CI :sweat_smile:

lwronski commented on modules/build/src/main/scala/scala/build/preprocessing/ExtractedDirectives.scala:
```diff
@@ -136,28 +128,11 @@ object ExtractedDirectives {
             StrictDirective(k.getPath.asScala.mkString("."), l.asScala.toSeq)
         }
 
-      val offset =
-        if (usedDirectives.getKind != UsingDirectiveKind.Code) 0
-        else usedDirectives.getCodeOffset
```
I think you are using the wrong offset below, for the `UsingDirectiveKind.SpecialComment` directive you should pass offset `0` into `ExtractedDirectives`, but you pass `specialCommentDirectives.getCodeOffset`.

I think you can remove the `offset` field from `ExtractedDirectives`, now it is no longer needed.

tgodzik replied:
Removed it!

lwronski reviewed (commented):
If you fix my comment, the unit tests should pass.

tgodzik commented on modules/build/src/test/scala/scala/build/tests/SourcesTests.scala:
```diff
@@ -245,27 +245,6 @@ class SourcesTests extends munit.FunSuite {
     }
   }
 
-  test("should fail dependencies in .java  with using keyword") {
```
I removed the test since code `using` can no longer be used

lwronski reviewed (approved):
LGTM, @tgodzik thanks!

tgodzik commented:
@lwronski the test failure looks unrelated or should I take a look?
